wagtailio/packages/views.py

```diff
@@ -0,0 +1,61 @@
+import requests
+from django.shortcuts import render
+from django.views import View
+from wagtail.admin import messages
+
+from .models import Grid, Package
+
+
+def process(url="https://djangopackages.org/api/v4/grids/?q=wagtail"):
+    gird_data = requests.get(url).json()
+    for item in gird_data.get("results", []):
+        title = item.get("title", "")
+        if "wagtail" in title.lower():
+            defaults = {key: item[key] for key in ["title", "slug", "description"]}
+            grid, _ = Grid.objects.update_or_create(
+                uid=item.get("id"), defaults=defaults
+            )
+            for idx, url in enumerate(item.get("packages", [])):
+                package_data = requests.get(url).json()
+                defaults = {
+                    key: package_data[key]
+                    for key in [
+                        "title",
+                        "slug",
+                        "repo_url",
+                        "pypi_version",
+                        "repo_forks",
+                        "repo_description",
+                        "pypi_url",
+                        "documentation_url",
+                        "repo_watchers",
+                        "participants",
+                    ] if not package_data[key] is None
+                }
+                defaults.update({"order": idx})
+                package, _ = Package.objects.update_or_create(
+                    uid=package_data.get("id"), defaults=defaults
+                )
+                grid.packages.add(package)
+
+    url = gird_data.get("next")
+    if url:
+        process(url)
+
+
+class IndexView(View):
+    http_method_names = ["get", "post"]
+
+    def post(self, request):
```

Comment:
worth wrapping this in a transaction in case the processing fails?

Reply:
Also, since it's making a lot of API requests, wrapping this in a transaction will make the update run atomically